app/services/harvest_weblog.rb

```
@@ -0,0 +1,114 @@
+require 'mysql2'
+require 'ipaddr'
+
+# Harvest a Web Logg, and save to mysql
+#
+class HarvestWeblogs
+  @db_connection = nil
+
+  #Util Class that parses a single line
+  class LineRecord
+    attr_accessor :agent, :event, :event_time, :ip, :method, :path, :pid, :status
+
+
+    # parse the line record
+    def initialize(line)
+      fields = line.split(' ')
+      @ip = fields[0]
+      @event = nil 
+      @status = fields[9]
+      @method = fields[6].sub('"','')
+      @path = fields[7]
+      @event_time = DateTime.strptime(fields[3].sub('[',''), "%d/%b/%Y:%H:%M:%S")
+      @pid  = nil
+      @agent = line
+    end
+
+    # save the record using the active record framework
+    def save
+      event = FedoraAccessEvent.new
+      create_time = DateTime.now
+      agent_format = agent.split('"')[5]
+      event.pid = pid
+      event.agent = agent_format
+      event.event = event
+      event.location = IPAddr.new(ip).mask(24).to_s.split("/")[0]
+      event.event_time = event_time
+      event.created_at = create_time
+      event.updated_at = create_time
+      event.save
```

If the record does not save (for some reason) do we want to raise an exception? Or is it okay to move along our merry way?